src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -383,6 +383,14 @@ mongoc_topology_new (const mongoc_uri_t *uri, bool single_threaded)
 #endif
 
    topology = (mongoc_topology_t *) bson_malloc0 (sizeof *topology);
+   // Check if requested to use TCP for SRV lookup.
+   {
+      char *MONGOC_SRV_PREFER_TCP = _mongoc_getenv ("MONGOC_SRV_PREFER_TCP");
+      if (MONGOC_SRV_PREFER_TCP && 0 == strcmp (MONGOC_SRV_PREFER_TCP, "ON")) {
```

Comment:
AFAIK, excluding tests, any/all env-var-based feature flags in libmongoc that affect runtime behavior only check if the env-var is (not) _defined_ without caring for its value (i.e. `MONGOC_DISABLE_SHM`). It may be preferable to apply that pattern here as well unless we expect non-boolean values to be used with this variable in the future:
```c
const bool prefer_tcp = _mongoc_getenv ("MONGOC_SRV_PREFER_TCP") != NULL;

if (prefer_tcp) {
  topology->srv_prefer_tcp = true;
}
```

Reply:
Updated to check for the presence of `MONGOC_EXPERIMENTAL_SRV_PREFER_TCP`.
Rest of suggestion was not accepted due to need to free the return of `_mongoc_getenv`.